zarr/tests/test_storage.py

```diff
@@ -1337,6 +1343,21 @@ def test_s3_complex(self):
         )
         assert (a[:] == -np.ones((8, 8, 8))).all()
 
+    def test_exceptions(self, memory_store):
+        path = memory_store.path
+        m = memory_store.fs
+        g = zarr.open_group(memory_store, mode="w")
+        arr = g.create_dataset("data", data=[1, 2, 3, 4], dtype="i4", compression=None, chunks=[2])
+        m.store[path + "/data/0"] = None
```

Comment:
Can there be a mock (maybe on fsspec.FSMap) to give a specific expected exception? Without that, the `raise` case in the changed code is not being tested, only the OK case, because a value is getting passed back.

Do we also need to test the single-item case?

Reply:
@itcarroll : any thoughts here?

Reply:
I can take a stab at the suggesiton next week. Thanks for the reminder.

Reply:
I think I did this without a mock by taking advantage of being able to "corrupt" the fsspec memory store: setting file objects to `None` rather than some bytes causes an `AttributeError` on reads which is not one of the FSStore.exceptions. The test now traverses both the `continue` case (silences its own `KeyError`) and the `raise` case (from the `AttributeError`) for `getitems`. It also covers the raise case for `__getitem__`, which gets called for scalar datasets.

Reply:
+1

Reply:
@joshmoore Could you authorize workflows, plz? Thank you!